server/src/main/java/org/opensearch/index/shard/IndexingStats.java

```diff
@@ -59,6 +63,98 @@ public class IndexingStats implements Writeable, ToXContentFragment {
      */
     public static class Stats implements Writeable, ToXContentFragment {
 
+        /**
+         * Tracks item level rest category class codes during indexing
+         *
+         * @opensearch.internal
+         */
+        public static class DocStatusStats implements Writeable, ToXContentFragment {
+
+            private final Map<Integer, AtomicLong> docStatusCounter;
```

Comment:
do we really need map, we need only 3 atomic variables right now, 2xx, 4xx, 5xx right